02.calendar/cal.rb

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'date'
+
+def calender
+  params = ARGV.getopts('m:', 'y:')
+  month = month(params['m'])
+  year = year(params['y'])
+
+  first_date = Date.new(year, month, 1)
+  last_date = Date.new(year, month, -1)
+
+  calender_dates = calender_dates(first_date, last_date)
+
+  carendar_display_datas = carendar_display_datas(first_date, calender_dates)
+
+  carendar_display_datas.each do |carendar_display_data|
+    puts carendar_display_data.join(' ')
+  end
+end
+
+def month(param_month)
+  param_month.nil? || !param_month.match?(/\A[0-9]+\z/) || !param_month.to_i.between?(1, 12) ? Date.today.month.to_i : param_month.to_i
```

Comment:
```suggestion
  param_month.to_i.between?(1, 12) ? param_month.to_i : Date.today.month
```

だけで要件は満たせるかも？

MEMO

- 条件分岐は否定形ではなく肯定形の条件にした方が読みやすいです
- monthは元から整数なのでto_iは不要です

Reply:
わざわざ`match?(/\A[0-9]+\z/) `のようにしなくても、`param_month.to_i`で数字じゃない場合は0に変換されるため問題ないことを確認いたしました！
MEMOに書いていただいた部分も意識して実装していきます！(特にto_iはとりあえずでつけてしまったので、レシーバのクラスが何なのかなどもちゃんを気をつけていきたいです)